### PR TITLE
docs: added flow for create_oauth2_client api

### PR DESCRIPTION
**Pull request by AdithyaSSup**
## Summary of change

Added flow for create oauth2_client api.

## Related issues

- https://github.com/supertokens/supertokens-core/issues/582

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here
highlighting the necessary changes)

## Checklist for important updates

- [ ] Changelog has been updated
    - [ ] If there are any db schema changes, mention those changes clearly
- [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
- [ ] `pluginInterfaceSupported.json` file has been updated (if needed)
- [ ] Changes to the version if needed
    - In `build.gradle`
- [ ] If added a new paid feature, edit the `getPaidFeatureStats` function in FeatureFlag.java file
- [ ] Had installed and ran the pre-commit hook
- [ ] If there are new dependencies that have been added in `build.gradle`, please make sure to add them
  in `implementationDependencies.json`.
- [ ] Issue this PR against the latest non released version branch.
    - To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the
      latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    - If no such branch exists, then create one from the latest released branch.

### CODE REVIEW

File: oauth2Flows.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    name: string,
+    scopes: string[]
+}
+```
+- Output:
+```
+{
+    status : "OK",
+    clientId: string,
+    clientSecretBase64: string,
```
Comment (rishabhpoddar) [inline oauth2Flows.md:18]: why is this returned as base64? When querying the core with the secret, is this to be sent as base64?

File: oauth2Flows.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Flow diagrams
```
Comment (rishabhpoddar) [inline oauth2Flows.md:1]: you can move this file to a folder called `flows` and we can keep it - this is good doc to have anyway

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: oauth2Flows.md (hunk 2 above)
Comment (AdithyaSSup) [reply oauth2Flows.md]: changes made , added todo's as well

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: flows/oauth2Flows.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+} | {
+    status : "CLIENT_ALREADY_EXISTS_ERROR"
+} | {
+    status : "UNKNOWN_SCOPE_ERROR",
+    scope : string
+}
+```
+- Flow diagram: https://app.code2flow.com/flowcharts/6475a4953f4c2482c82a4352
+
+## 2. Get oauth2 client  [in progress]
```
[the remark this answers is not in the document]
Comment (AdithyaSSup) [reply flows/oauth2Flows.md:32]: changes made

File: flows/oauth2Flows.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+```
+- Flow diagram: https://app.code2flow.com/flowcharts/6475cc5b3f4c2482c82a736a
+
+## Todo's:
+- [ ] Add implementation of `upsertOAuth2ClientScope_Transaction()` in postgres-plugin
+- [ ] Update changes to Create oauth2 client api contract in `core-driver-interface` 
+- [ ] Add `create_oauth2_client` api implementation in `supertokens-core`
+- [ ] Add test cases for create `oauth2_client` api  in `supertokens-core`
```
[the remark this answers is not in the document]
Comment (AdithyaSSup) [reply flows/oauth2Flows.md]: changes made

**Pull request merged**